feat(assert): cdk assert: implement 'haveOutput' assertion (#1906) (#5366)
aws/aws-cdk: packages/@aws-cdk/assert/README.md

    // Note: some properties omitted here
}));
```
### Check existence of an output
`haveOutput` assertion can be used to check that a stack contains specific output. 
Parameters to check against can be:

> - output name

Reviewer: ```suggestion
- `outputName`
```

> - output value

Reviewer: ```suggestion
- `outputValue`
```

> - export name

Reviewer: ```suggestion
- `exportName`
```

> If output value is provided, at least one of \[output name, export name\] should be proviled as well

Reviewer: ```suggestion
If `outputValue` is provided, at least one of `outputName`, `exportName` should be provided as well
```

Reviewer: Also - the code suggests either `outputName` or `exportName` is required **regardless** of whether `outputValue` is provided or not.

Example
```ts
expect(synthStack).to(haveOutput({
  outputName: 'TestOutputName',
  exportName: 'TestOutputExportName',
  outputValue: {
    'Fn::GetAtt': [
      'TestResource',
      'Arn'
    ]
  }
}));

> 

Reviewer: Remove blank line.
